fluentd-gcp-scaler/scaler.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/sh
+
+SLEEP_SECONDS=${SLEEP_SECONDS:-60}
+NAMESPACE=${NAMESPACE:-kube-system}
+DEFAULT_CPU_REQUEST=${CPU_REQUEST}
+DEFAULT_MEMORY_REQUEST=${MEMORY_REQUEST}
+DEFAULT_CPU_LIMIT=${CPU_LIMIT}
+DEFAULT_MEMORY_LIMIT=${MEMORY_LIMIT}
+
+reset_to_defaults() {
+  REQUESTS=
+  LIMITS=
+  CPU_REQUEST=${DEFAULT_CPU_REQUEST}
+  MEMORY_REQUEST=${DEFAULT_MEMORY_REQUEST}
+  CPU_LIMIT=${DEFAULT_CPU_LIMIT}
+  MEMORY_LIMIT=${DEFAULT_MEMORY_LIMIT}
+}
+
+apply_scaling() {
+  if ! kubectl get scalingpolicies -n ${NAMESPACE} ${SCALING_POLICY}
+  then
+    echo "${SCALING_POLICY} not found in namespace ${NAMESPACE}, using defaults."
+    exit 0
+  fi
+  for resource_class in request limit
+  do
+    for resource_type in cpu memory
+    do
+      TMP=$(kubectl get scalingpolicies -n ${NAMESPACE} ${SCALING_POLICY} \
+        -o=jsonpath="{.spec.containers[?(@.name=='fluentd-gcp')].resources.${resource_class}s[?(@.resource=='${resource_type}')].base}")
```

Comment:
Can you pass a DS name via flag/env var?

Reply:
DS name is passed by a env var. 'fluentd-gcp' here is a container name and I don't see a reason to set it externally.